src/components/ErrorSnackbar.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useEffect } from 'react';
+import { Snackbar, Alert } from '@mui/material';
+
+const ErrorSnackbar = () => {
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    if (window.electronAPI) {
```

Comment:
I have tried for many hours to make Redux work with IPC interactions, but it still did not work: Redux was able to update the store of the Error statuses (needed for the snackbar), but the component did not receive any update of such store, mostly because of the different natures of the IPC interactions and Redux state management.

I find it more reasonable to centralize IPC interactions in dedicated modules (e.g., `src/electron/electronApi.ts` and `src/cosimulation/cosimulationApi.ts`) and update the UI directly through events or callbacks, bypassing Redux for these interactions.
Redux could still then be reserved for UI-related state that doesn’t depend on Electron.

Is there something missing, or maybe a way to achieve communication between both? I also have tried to use different libraries, but most of them are outdated/deprecated or they do not have any support anymore.